DIPs/DIP1002.md

```diff
@@ -0,0 +1,90 @@
+# TryElseExpression
+
+| Field           | Value                                                           |
+|-----------------|-----------------------------------------------------------------|
+| DIP:            | 1002                                                            |
+| Author:         | [Sophie Kirschner](https://github.com/pineapplemachine)         |
+| Implementation: | n/a                                                             |
+| Status:         | Draft                                                           |
+
+## Abstract
+
+In Python, the try/catch/finally syntax is augmented with an additional clause,
+termed else. It is a fantastically useful addition to the conventional syntax.
+It works like this:
+
+``` python
+    try:
+        do_something()
+    except Exception as e:
+        pass # Runs when an error inheriting from Exception was raised
+    else:
+        pass # Runs when no error was raised
+    finally:
+        pass # Runs unconditionally, evaluates last
+```
+
+### Links
+
+* [Exception handling in Python](https://docs.python.org/3/tutorial/errors.html#handling-exceptions)
+
+## Description
+
+The _TryStatement_ would gain an additional piece, _TryElseStatement_, which
+should be present after all _Catches_ and before a _FinallyStatement_, if any.
+
+The code in a _TryElseStatement_ would occur after the try's _ScopeStatement_,
+would occur only if no exception was thrown, and would occur before the
+_FinallyStatement_, if any is present. If any error occurs inside a
+_TryElseStatement_, its corresponding _FinallyStatement_ is still evaluated.
+
+### Rationale
+
+Having such a facility built in to D's exception handling will allow the writing
```

Comment:
@schancel Here is an example in puesdo-code reduced from a real world example. It's a project that reads values from sensors and then saves them to a server

```Python
        try:
            for sensor in sensor_objects:
                get_value()  # this can throw
                if value is not None and value != 'NULL':
                    save_value()  # this can throw

            change_notifier_state()
        except IOError:
            change_notifier_state()
        except KeyboardInterrupt:
            change_notifier_state()
            exit()
        except (SeverException, ConnectionException) as e:
            # if there's server trouble, wait longer to
            # avoid DDOSing the server
            time.sleep(45)
            change_notifier_state()
        else:
            time.sleep(5)
```

As noted in the DIP, this can be achieved without the else clause, but it makes things clearer.